core/src/components/item-sliding/test/async/index.html

```diff
@@ -23,25 +23,56 @@
       </ion-header>
 
       <ion-content>
+        <ion-button id="toggle-button" onclick="toggle()">Make async changes</ion-button>
         <ion-list>
-          <ion-item-sliding>
+          <ion-item-sliding id="async-item">
+            <ion-item-options side="end">
+              <ion-item-option>Option</ion-item-option>
+            </ion-item-options>
+          </ion-item-sliding>
+          <ion-item-sliding id="async-options-added">
+            <ion-item>ion-item-options added async</ion-item>
+          </ion-item-sliding>
+          <ion-item-sliding id="async-options-removed">
+            <ion-item>ion-item-options removed async</ion-item>
             <ion-item-options side="end">
               <ion-item-option>Option</ion-item-option>
             </ion-item-options>
           </ion-item-sliding>
         </ion-list>
 
         <script>
-          const itemSliding = document.querySelector('ion-item-sliding');
-          setTimeout(() => {
-            const item = document.createElement('ion-item');
-            item.innerText = 'Item Sliding Option';
-            itemSliding.appendChild(item);
+          const slidingAsyncItem = document.querySelector('#async-item');
+          const slidingAsyncOptionsAdded = document.querySelector('#async-options-added');
+          const slidingAsyncOptionsRemoved = document.querySelector('#async-options-removed');
 
-            item.onclick = () => {
+          [slidingAsyncOptionsAdded, slidingAsyncOptionsRemoved].forEach((itemSliding) => {
+            itemSliding.querySelector('ion-item').onclick = () => {
               itemSliding.open('end');
             };
-          }, 250);
+          });
+
+          /**
+           * Function used rather than setTimeout to ensure ion-item-slidings
+           * are fully inited before making changes. Otherwise some bugs would
+           * not appear or be flaky.
+           */
+          function toggle() {
```

Comment:
This does something funky if you press the toggle button multiple times

https://github.com/ionic-team/ionic-framework/assets/2721089/f929721e-521d-4316-a6ba-eddfda4a5eea